src/sh/python-lint/run-in-poetry-subdir.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+# cspell:words realpath, autoflake, venv, toplevel
+
+# Capture the first argument, which is the command we're wrapping for
+# the pre-commit hook.
+COMMAND=$1
+# Then skip it so we can pass "$@" to the command.
+shift
+
+# Capture and skip like above.
+ERROR_MESSAGE=$1
+shift
+
+RELATIVE_PATHS=""
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+PYTHON_DIR=$ROOT_DIR/src/python
+POETRY_SUBDIRECTORY=$PYTHON_DIR/hooks
+
+# Convert all paths to relative paths.
```

Comment:
Noting here that apparently Apple overloads `realpath` and so it might be more effective to do this operation in Python